aptos-move/block-executor/src/executor.rs

```diff
@@ -623,11 +606,60 @@ where
             // TODO: re-use the code cache.
             drop(versioned_cache);
         });
+        ret
+    }
 
+    fn make_final_result_in_parallel(
```

Comment:
I think this may be done more nicely as a part of commit_epilogue, maybe @ziaptos can see if that can be added here? 
https://github.com/aptos-labs/aptos-core/pull/10081

Reply:
If it can be done cleanly in the refactor and that's something we can land soon, then I am fine abandoning this in favor of including it in the refactor. Otherwise, we can land this change and see if this can be done more nicely as a part of ongoing refactor? Thoughts?

Reply:
I would be in favor of landing this and trying it out. We can do a proper refactoring only if we see value.

Reply:
I don't see an issue to refactor as needed after landing it, but I think we should land if we see value? We can check the effect already on the PR, right?

Reply:
@gelash - Yes we are seeing around 4-5% TPS boost with executor benchmark.

Reply:
Good point. I see we are already discussing the improvements in the slack thread.

Reply:
so, once we have the numbers, happy to approve as is

Reply:
Updated the test plan with the recent numbers.

Reply:
I'm OK with landing this independently if we see value - tho further refactoring can bring more value.

Reply:
With the determinism issue, it does seem a lot to do it this way if indeed there is be a much cleaner (and more efficient, but not sure how noticeable) solution. And I feel more so after accepting and realizing I didn't think of smt I should have (and 99% even ran into the same thing long time ago).. So I am inclined to not rush it now, unless we really need this on the main asap, I'd rather us try it the other way? (coordinator goes sequentially and can stop the execution when it sees first abort/skiprest, we already do so for gas, workers can do the other part - and they are parallelized and pipelined with main work already, without spawning anew - unless I'm missing smt this seems to be what we need thanks to the rolling commit? - I don't think we had revisited this after rolling commit)

Reply:
@gelash - Indeed the current approach is a bit convoluted to maintain the determinism. I am okay with waiting to land the block-stm refactor and doing it in a cleaner and potentially more efficient way. However, I would like to get this optimization in main in a week or so, so that we can include this in previewnet. Do you think that's a reasonable timeline? If not, I would be inclined to land the current change as a temporary approach, while the block-stm refactor is making progress.